react-app/src/components/SaveOutputForm.js

```diff
@@ -0,0 +1,213 @@
+
+  // let whichFiles = {};
+
+  function togglePatientCheckbox(e) {
+    setWhichFiles({ ...whichFiles, [e.target.id]: e.target.checked });
+    if (!e.target.checked && selectAll) {
+      setSelectAll(false);
+    }
+  }
+
+  function getPatientCheckboxes() {
+    const list = props.extractedData.map((bundle, i) => {
+      const label = getLabel(bundle, i);
+      return (
+        <Form.Check
+          type="checkbox"
+          label={label}
+          key={label}
+          id={label}
+          checked={whichFiles[label]}
+          onChange={togglePatientCheckbox}
+        />
+      );
+    });
+    return list;
+  }
+
+  function toggleSelectAll() {
+    // modify which files will be saved first, so that there's no issues with state update delays
+    if (!selectAll) {
+      props.extractedData.forEach((bundle, i) => {
+        const label = getLabel(bundle, i);
+        defaultWhichFiles = {
+          ...defaultWhichFiles,
+          [label]: true,
+        };
+      });
+      setWhichFiles({ ...defaultWhichFiles });
+    }
+    // set select all
+    setSelectAll(!selectAll);
+  }
+
+  function onSave() {
+    const outputBundles = [];
+    props.extractedData.forEach((bundle, i) => {
+      if (whichFiles[getLabel(bundle, i)] === true) {
+        outputBundles.push({ bundle, index: i });
+      }
+    });
+    if (outputBundles.length > 0) {
+      window.api
+        .saveOutput(outputPath, outputBundles, props.loggedMessages, saveLogs)
+        .then((result) => {
+          if (result === true) {
+            // if saveOutput() returns true, then the save process succeeded
+            setShowSavedAlert(true);
+          }
+        })
+        .catch((error) => {
+          setShowErrorAlert(true);
+          setErrorMessage(error.message);
+          setShowSavedAlert(false);
+          setShowNoFilesAlert(false);
+        });
+    } else {
+      setShowNoFilesAlert(true);
+    }
+  }
+  function onCancel() {
+    props.setShowSaveForm(false);
+  }
+  return (
+    <div className="page-container">
+      <h3 className="page-subtitle">Save Output</h3>
+      <div className="page-container">
+        <Form className="form-container">
+          <Row>
+            <Col>
+              <Form.Group controlId="formConfigPath" className="mb-3">
+                <Form.Label className="form-label">Select Output Folder</Form.Label>
+                <div className="file-picker-box">
+                  <div className="file-button-container">
+                    <Button className="generic-button narrow-button" variant="outline-info" onClick={onSetOutputPath}>
+                      Select Folder
+                    </Button>
+                    <Button className="generic-button narrow-button" variant="outline-info" onClick={clearOutputPath}>
+                      Clear
+                    </Button>
+                  </div>
+                  <Form.Label className="form-label file-name">{outputPath}</Form.Label>
```

Comment:
The separate input location for choosing where to save the files threw me off a little bit. When I used the app, I found that I was expecting the "Select Folder" button inside the dialog box to save the files, rather than just choose the right file path. I think this is because there are so many other "Save As" workflows in other apps. I think it might make sense to move this functionality to just the "Save" button in this component, and relabel that button to be a "Save As" button a user expects to navigate to a directory in the dialog box. What do you think of that proposal?

Reply:
I have moved the folder selection functionality to the Save button, which has been renamed to Save As.

Reply:
I don't really think "Save As" is the right label for this button either.... 
"Select Folder and Save"? 
"Save to Folder"?
Which of those two resonate most with you two?